lib/codequest_pipes/context.rb

```diff
@@ -73,9 +70,37 @@ def failure?
     def inspect
       keys = methods - Object.methods - Pipes::Context.instance_methods
       fields = keys.map { |key| "#{key}=#{public_send(key).inspect}" }
-      fields << "@error=#{@error.inspect}"
+      fields << "@errors=#{@errors.inspect}"
       object_id_hex = '%x' % (object_id << 1)
       "#<Pipes::Context:0x00#{object_id_hex} #{fields.join(', ')}>"
     end
+
+    # Return errors from ErrorCollector object.
+    #
+    # @return [Hash]
+    def errors
+      error_collector.errors
+    end
+
+    # This method is added to maintain backwards compatibility - previous
+    # versions implemented a single @error instance variable of String for error
+    # storage.
+    #
+    # @return [String]
+    def error
```

Comment:
Maybe we should add `deprecated` warning here?

Reply:
We will not do that, since this PR will be released in new non-backward-compatible version.